.github/workflows/hash_collisions.yml

```diff
@@ -0,0 +1,54 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+name: Rust Hash Collisions
+
+concurrency:
+  group: ${{ github.repository }}-${{ github.head_ref || github.sha }}-${{ github.workflow }}
+  cancel-in-progress: true
+
+# https://docs.github.com/en/actions/writing-workflows/choosing-when-your-workflow-runs/events-that-trigger-workflows#running-your-pull_request-workflow-when-a-pull-request-merges
+#
+# this job is intended to only run on merge to main branch
+on:
+  pull_request:
+    branches:
+      - main
+    types:
+      - closed
```

Comment:
The way we do it in arrow is

```yaml
on:
  # always trigger
  push:
    branches:
      - main
```

https://github.com/apache/arrow-rs/blob/4a0bdde24f48d0fc1222d936493f798d9ea4789d/.github/workflows/arrow.yml#L25-L29

I'll make a small follow on PR to touch it up if necessary